controller/pkg/extensions/konnect/controlplane.go

```diff
@@ -28,7 +60,126 @@ func ApplyControlPlaneKonnectExtension(ctx context.Context, cl client.Client, co
+				ControlPlaneID: konnectExtension.Status.Konnect.ControlPlaneID,
+				TLSClient: managercfg.TLSClientConfig{
+					Cert: TLSClientCert,
+					Key:  TLSClientKey,
+				},
+				LicenseSynchronizationEnabled: true,
+				ConfigSynchronizationEnabled:  true,
+				UploadConfigPeriod:            time.Second * 10,
+				InitialLicensePollingPeriod:   time.Second * 10,
+				LicensePollingPeriod:          time.Second * 10,
+			},
+		}, nil
+
+	case konnectv1alpha2.ClusterTypeControlPlane:
+		return nil, fmt.Errorf("unsupported Konnect cluster type: %s", konnectExtension.Status.Konnect.ClusterType)
+	default:
+		// default never happens as the validation is at the CRD level
+		panic(fmt.Sprintf("unsupported Konnect cluster type: %s", konnectExtension.Status.Konnect.ClusterType))
+	}
+}
+
+// buildKonnectAddress builds the Konnect address out of the control plane endpoint.
+// input: "https://7b46471d3b.us.tp0.konghq.tech:443"
+// output: "https://us.kic.api.konghq.tech"
+func buildKonnectAddress(endpoint string) string {
+	portlessEndpoint := strings.TrimSuffix(endpoint, ":443")
+	addressSlice := strings.Split(portlessEndpoint, ".")
+	addressSlice = lo.Filter(addressSlice, func(_ string, i int) bool {
+		// remove 7b46471d3b and tp0
+		if i == 0 || i == 2 {
+			return false
+		}
+		return true
+	})
+	return fmt.Sprintf("https://%s.kic.api.%s", addressSlice[0], strings.Join(addressSlice[1:], "."))
+}
+
+// getTLSClientCertAndKey retrieves the TLS client certificate and key from the specified secret.
+// Returns the certificate and key as strings, or an error if the secret is not found or if the data is missing.
+// The secret must contain "tls.crt" and "tls.key" keys.
+// This function is used to extract the TLS client cert and key for Konnect configuration.
+func getTLSClientCertAndKey(ctx context.Context, cl client.Client, secretName, secretNamespace string) (tlsClientCert, tlsClientKey string, err error) {
+	secret := &corev1.Secret{}
+	if err := cl.Get(ctx, client.ObjectKey{Namespace: secretNamespace, Name: secretName}, secret); err != nil {
+		return "", "", fmt.Errorf("failed to get TLS client secret %s/%s: %w", secretNamespace, secretName, err)
+	}
+
+	tlsCert, ok := secret.Data["tls.crt"]
+	if !ok {
+		return "", "", fmt.Errorf("TLS certificate not found in secret %s/%s", secretNamespace, secretName)
+	}
+
+	tlsKey, ok := secret.Data["tls.key"]
+	if !ok {
+		return "", "", fmt.Errorf("TLS key not found in secret %s/%s", secretNamespace, secretName)
+	}
+
+	return string(tlsCert), string(tlsKey), nil
+}
+
+// applyFeatureGatesToControlPlane applies the FillIDs feature gate to the ControlPlane's FeatureGates.
+// If the FeatureGates field is nil, it initializes it with the FillIDs feature gate.
+// If the FillIDs feature gate is already present, it ensures its value is set to enabled.
+// If the FillIDs feature gate is not present, it appends it to the FeatureGates slice.
+func applyFeatureGatesToControlPlane(cp *gwtypes.ControlPlane) {
```

Comment:
Looks like the function only adds the `fillIDsFeatureGate` to the CP but not other feature gates. Could the name `applyFillIDsFeatureGatesToControlPlane` describe it better?